lib/cvprac/api/configlet.rb

```diff
@@ -0,0 +1,304 @@
+# encoding: utf-8
+# BSD 3-Clause License
+#
+# Copyright (c) 2017, Arista Networks EOS+
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name Arista nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# @author Arista EOS+ Consulting Services <eosplus-dev@arista.com>
+module Cvprac
+  # CVP Configlet api methods
+  module Api
+    # CVP Configlet api methods
+    # rubocop:disable Metrics/ModuleLength
+    module Configlet
+      # @!group Configlet Method Summary
+
+      # Add configlet
+      #
+      # @param [String] name The name of the desired configlet
+      # @param [String] config Multiline string of EOS configuration
+      #
+      # @return [String, nil] The key for the new configlet. nil on failure
+      #
+      # @raise CvpApiError on failure.  Common: errorCode: 132518: Data already
+      #   exists in Database.
+      #
+      # @example
+      #    result = api.add_configlet('api_test_3',
+      #                               "interface Ethernet1\n   shutdown")
+      def add_configlet(name, config)
+        log(Logger::DEBUG) do
```

Comment:
Having the conditional around each log statement is wierd. We usually just take the hit on the function call to reduce code when logging. Why not log(Logger::DEBUG, 'message string') ?